File: prob10.m
function [ out ] = prob10( n )

a = n*2;
arr = [1:a];

arr(4:2:a) = 0;

for x = [3:2:a];
    y = 2*x;
    new = [y:x:a];
    arr(new) = 0;
end
    arr(1) = [];
    arr(arr == 0) = [];
    arr(arr>n) = [];
    out = sum(arr(arr<n+1));
end

File: prob10.py
import math

def findPrimes(num):
    primeList = range(num + 1)
    i = 2
    while i < num ** 0.5:
        while primeList[i] == 0:
            i += 1
        j = primeList[i]
        while j < num:
            j += i
            if j < num:
                primeList[j] = 0
        i += 1
    return primeList

primes = findPrimes(10000000)
print sum(primes)
